pkg/asset/manifests/operators.go

```diff
@@ -318,62 +316,41 @@ func (m *Manifests) Load(f asset.FileFetcher) (bool, error) {
 }
 
 func redactedInstallConfig(config types.InstallConfig) ([]byte, error) {
-	newConfig := config
+	// Use DeepCopy to create a proper deep copy, avoiding the shallow copy issue
+	// where nested pointer fields would still reference the original struct.
+	newConfig := config.DeepCopy()
 
+	// Redact top-level sensitive fields
 	newConfig.PullSecret = ""
+
+	// Redact platform-specific sensitive fields
+	// Platforms are mutually exclusive, so we use a switch statement
 	switch {
 	case newConfig.Platform.VSphere != nil:
-		p := config.VSphere
-		newVCenters := make([]vsphere.VCenter, len(p.VCenters))
-		for i, v := range p.VCenters {
-			newVCenters[i].Server = v.Server
-			newVCenters[i].Datacenters = v.Datacenters
-		}
-		newVSpherePlatform := vsphere.Platform{
-			DeprecatedVCenter:          p.DeprecatedVCenter,
-			DeprecatedUsername:         "",
-			DeprecatedPassword:         "",
-			DeprecatedDatacenter:       p.DeprecatedDatacenter,
-			DeprecatedDefaultDatastore: p.DeprecatedDefaultDatastore,
-			DeprecatedFolder:           p.DeprecatedFolder,
-			DeprecatedCluster:          p.DeprecatedCluster,
-			DeprecatedResourcePool:     p.DeprecatedResourcePool,
-			ClusterOSImage:             p.ClusterOSImage,
-			DeprecatedAPIVIP:           p.DeprecatedAPIVIP,
-			APIVIPs:                    p.APIVIPs,
-			DeprecatedIngressVIP:       p.DeprecatedIngressVIP,
-			IngressVIPs:                p.IngressVIPs,
-			DefaultMachinePlatform:     p.DefaultMachinePlatform,
-			DeprecatedNetwork:          p.DeprecatedNetwork,
-			DiskType:                   p.DiskType,
-			VCenters:                   newVCenters,
-			FailureDomains:             p.FailureDomains,
+		// Redact deprecated credentials
+		newConfig.Platform.VSphere.DeprecatedUsername = ""
+		newConfig.Platform.VSphere.DeprecatedPassword = ""
+
+		// Redact VCenter credentials while preserving other fields
+		for i := range newConfig.Platform.VSphere.VCenters {
+			newConfig.Platform.VSphere.VCenters[i].Username = ""
+			newConfig.Platform.VSphere.VCenters[i].Password = ""
```

Comment:
```suggestion
			newConfig.Platform.VSphere.VCenters[i].Password = ""
			newConfig.Platform.VSphere.VCenters[i].Port = 0
```

I think previously, only `server` and `datacenters` were preserved. So, we also need to redact the `port` field.

We can see the problem reported by the unit test fail: [ci/prow/unit](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_installer/9972/pull-ci-openshift-installer-main-unit/1998734824977731584)